src/main/java/com/mabl/integration/jenkins/MablRestApiClientImpl.java

```diff
@@ -223,6 +259,23 @@ public String getAppBaseUrl() {
         return appBaseUrl;
     }
 
+    @Override
+    public void checkConnection() throws IOException {
+        final String url = restApiBaseUrl + HEALTH_ENDPOINT;
```

Comment:
I don't think we should use this endpoint. It's meant for internal checks and is quite expensive (makes calls to internal resources to determine overall API health). If we have a need to determine the API is up, I think we should choose a different endpoint. 

A liveness check endpoint would be a better candidate (e.g. is internal NOOP, `/health/live`).